src/libmongoc/src/mongoc/mongoc-stream-tls-secure-channel.c

```diff
@@ -974,24 +974,23 @@ mongoc_stream_tls_secure_channel_new (mongoc_stream_t *base_stream,
    schannel_cred.dwFlags = SCH_USE_STRONG_CRYPTO;
 #endif
 
+   /* By default, enable soft failing.
+    * A certificate with no revocation check is a soft failure. */
+   schannel_cred.dwFlags |= SCH_CRED_IGNORE_NO_REVOCATION_CHECK;
+   /* An offline OCSP responder / CRL distribution list is a soft failure. */
+   schannel_cred.dwFlags |= SCH_CRED_IGNORE_REVOCATION_OFFLINE;
```

Comment:
This looks fine to me. My main concern is does this work? Did you test against both a down responder and a responder sending "STATUS_UNKNOWN"?

Reply:
I tested against a down responder before, but good thing you reminded me about STATUS_UNKNOWN. The spec tests currently do not test that scenario.

I tested the scenarios on Windows by following these steps:

Starting a mongod on Windows with no stapling using the RSA cert  from drivers-evergreen-tools.

Starting the mock responder in drivers-evergreen-tools:

```
python.exe ..\ocsp_mock.py  --ca_file ca.pem --ocsp_responder_cert ca.crt --ocsp_responder_key ca.key -p 8100 -v --fault unknown
```

Flushing the OCSP response cache with:
```
certutil -urlcache * delete
certutil -flushcache 0 1
```

And verifying connection succeeded. I repeated the same with `--fault unknown` to verify connection failed. And did the same with no responder running to ensure offline is a soft-fail.

Reply:
Cool looks like soft-fail works! Also in the last paragraph I assume you mean `--fault revoked`?

Reply:
Ah yes, that was a typo, I meant `--fault revoked`.